In [1]:
import pandas as pd
import numpy as np
bollywood_celebrities = pd.read_csv('bollywood_celebrities.csv')['celeb'].values.tolist()
slangs = ['wtf','omg','lit','flex','mood','unreal','lol','meme','oops','lmao','rofl']
porn_words = ['mms','porn','hot','sexy','cleavage','sex','oops','blue film','nude','xxx','kiss','bikini','bed scene','18+','adult','dirty','item song','watch alone','shower','naked']
celebrity_related_words = ['bollywood','controversial','dark side','death','deleted scene','celeb','diva','bolly','drunk','drugs','caught','nepotism','star kid','pregnant','fight','boycott','shameful','marriage','transformation','plastic surgery','live in relationship','scandal','affair','casting couch','blockbuster','superhit','flop']
clickbait_phrases = ['shocking','you wont believe','leave you speechless','sad news','astonishing','surprising','reasons why','things you','reveals','this is why','this is what','this is the','this is how','you can now','you will ever need','why you should','live','see','why we love','secret tips','best','worst','most','ultimate guide to','unbelievable','amazing','check out','will make you','tears of joy','make you cry','give you goosebumps','is too cute','melt your heart','can’t stop laughing','what happened next','twitter reacts to','are freaking out','harsh truths','the reason is','can we guess','things only','goes viral','unpleasant','unseen','then and now','emotional','embarassing','cringiest']

In [2]:
data = pd.read_csv('BollyBAIT.csv')
data.head()

Title  \
0  Brahmastra Trailer | Ranbir Kapoor | Alia Bhat...   
1  83 Movie Trailer | Ranveer Singh | Deepika Pad...   
2  Bell Bottom Trailer | Akshay Kumar | Jackky  B...   
3  Mahabharat - Official Trailer | Aamir Khan | H...   
4  Nick Jonas की इस हरकत पर भड़की Priyanka Chopra,...   

                                    Translated Title     Video ID  \
0  Brahmastra Trailer | Ranbir Kapoor | Alia Bhat...  zR8x5FAvlPU   
1  83 Movie Trailer | Ranveer Singh | Deepika Pad...  STL-ueSScRI   
2  Bell Bottom Trailer | Akshay Kumar | Jackky Bh...  bJ-1QOOett0   
3  Mahabharat - Official Trailer | Aamir Khan | H...  qfr0NUD3FTU   
4  Nick erupted on the movement of Jonas Priyanka...  D_TP0L-s4V0   

                                     Video URL Channel Title Misleading Video  \
0  https://www.youtube.com/watch?v=zR8x5FAvlPU  PR Bollywood               No   
1  https://www.youtube.com/watch?v=STL-ueSScRI  PR Bollywood               No   
2  https://www.youtube.com/watch?v=bJ-1QOOett0  PR Bollywood              Yes   
3  https://www.youtube.com/watch?v=qfr0NUD3FTU  PR Bollywood               No   
4  https://www.youtube.com/watch?v=D_TP0L-s4V0  PR Bollywood              Yes   

  False Promises  Exaggerated Video Spam Content  Exploits Curiosity Gap   \
0             Yes               Yes            No                      No   
1             Yes               Yes            No                      No   
2             Yes               Yes            No                      No   
3             Yes               Yes            No                      No   
4             Yes               Yes           Yes                      No   

       Label  Count of Celebrity Mentions  Count of Slangs  \
0  Clickbait                            3                0   
1  Clickbait                            0                0   
2  Clickbait                            0                0   
3  Clickbait                            4                0   
4  Clickbait                            1                0   

   Count of Porn Words  Count of Celebrity Related Words  \
0                    0                                 0   
1                    0                                 0   
2                    0                                 0   
3                    0                                 0   
4                    0                                 0   

   Count of Clickbait Phrases  
0                           0  
1                           0  
2                           0  
3                           0  
4                           0

In [4]:
from google_trans_new import google_translator  
#translator = google_translator() 

count_celebs = list()
count_slangs = list()
count_porn_words = list()
count_celebrity_related_words = list()
count_clickbait_phrases = list()
ids = list()

for index, row in data.iterrows():
    title = row['Translated Title'].lower()
        #print(titlex)
        #title = translator.translate(titlex,lang_tgt='en') 
    ids.append(row['Video ID'])

    count_celeb = 0
    count_slang = 0
    count_porn_word = 0
    count_celebrity_related_word = 0
    count_clickbait_phrase = 0

    for i in range(len(bollywood_celebrities)):
        if bollywood_celebrities[i] in str(title):
            count_celeb = count_celeb + 1
    count_celebs.append(count_celeb)

    for i in range(len(slangs)):
        if slangs[i] in str(title):
            count_slang = count_slang + 1
    count_slangs.append(count_slang)

    for i in range(len(porn_words)):
        if porn_words[i] in str(title):
            count_porn_word = count_porn_word + 1
    count_porn_words.append(count_porn_word)

    for i in range(len(celebrity_related_words)):
        if celebrity_related_words[i] in str(title):
            count_celebrity_related_word = count_celebrity_related_word + 1
    count_celebrity_related_words.append(count_celebrity_related_word)

    for i in range(len(clickbait_phrases)):
        if clickbait_phrases[i] in str(title):
            count_clickbait_phrase = count_clickbait_phrase + 1
    count_clickbait_phrases.append(count_clickbait_phrase)
    
dict = {'Count of Celebrity Mentions' : count_celebs, 'Count of Slangs' : count_slangs, 'Count of Porn Words' : count_porn_words, 'Count of Celebrity Related Words' : count_celebrity_related_words, 'Count of Clickbait Phrases' : count_clickbait_phrases}
newdf = pd.DataFrame(dict)

alia bhatt
amitabh bachchan
ranbir kapoor
deepika padukone
ranveer singh
akshay kumar
jackky bhagnani
deepika padukone
aamir khan
hrithik roshan
prabhas
priyanka chopra
alia bhatt
amitabh bachchan
ranbir kapoor
disha patani
randeep hooda
salman khan
sanjay dutt
deepika padukone
hrithik roshan
rakesh roshan
deepika padukone
hrithik roshan
rakesh roshan
pooja hegde
prabhas
veena
raveena tandon
sanjay dutt
aamir khan
ranbir kapoor
ranveer singh
kareena kapoor
aamir khan
vijay
anushka sharma
kareena kapoor
hrithik roshan
sunny leone
emraan hashmi
kareena kapoor
anushka sharma
anushka sharma
anushka sharma
anushka sharma
deepika padukone
deepika padukone
deepika padukone
deepika padukone
shraddha kapoor
fawad khan
katrina kaif
sidharth malhotra
jacqueline fernandez
kriti sanon
akshay kumar
priyanka chopra
amitabh bachchan
hrithik roshan
rakesh roshan
priyanka chopra
shruti haasan
alia bhatt
priyanka chopra
akshay kumar
govinda
sara ali khan
deepika padukone
anushka sharma
deepika padukone
a

nawazuddin siddiqui
disha patani
aditya roy kapur
anil kapoor
karan johar
rajkummar rao
vidya balan
alia bhatt
ranveer singh
aamir khan
pran
saif ali khan
priyanka chopra
amitabh bachchan
chiranjeevi
ram charan
amitabh bachchan
chiranjeevi
ram charan
alia bhatt
ranveer singh
alia bhatt
ranveer singh
katrina kaif
abhay deol
farhan akhtar
hrithik roshan
akshay kumar
akshay kumar
akshay kumar
ali fazal
manjot singh
pulkit samrat
varun sharma
kareena kapoor
shah rukh khan
deepika padukone
priyanka chopra
diljit dosanjh
alia bhatt
karan johar
varun dhawan
salman khan
kareena kapoor
priyanka chopra
priyanka chopra
deepika padukone
anushka sharma
kajol
ajay devgn
deepika padukone
anushka sharma
taapsee pannu
deepika padukone
priyanka chopra
deepika padukone
ananya panday
shraddha kapoor
sara ali khan
disha patani
shah rukh khan
shah rukh khan
shah rukh khan
shah rukh khan
shah rukh khan
shah rukh khan
deepika padukone
shah rukh khan
anushka sharma
shah rukh khan
aamir khan
tara sutaria
ananya

In [5]:
dict = {'IDS': ids, 'Count of Celebrity Mentions' : count_celebs, 'Count of Slangs' : count_slangs, 'Count of Porn Words' : count_porn_words, 'Count of Celebrity Related Words' : count_celebrity_related_words, 'Count of Clickbait Phrases' : count_clickbait_phrases}
newdf = pd.DataFrame(dict)
newdf
#print(len(count_celebs))

IDS  Count of Celebrity Mentions  Count of Slangs  \
0    zR8x5FAvlPU                            3                0   
1    STL-ueSScRI                            2                0   
2    bJ-1QOOett0                            2                0   
3    qfr0NUD3FTU                            4                0   
4    D_TP0L-s4V0                            1                0   
..           ...                          ...              ...   
991  MBrAY6GgU5k                            2                0   
992  owPuQjInzO8                            2                0   
993  Es2bli0uSYY                            0                0   
994  FgHKdoYpGpI                            1                0   
995  PJ-mIEZMJes                            2                0   

     Count of Porn Words  Count of Celebrity Related Words  \
0                      0                                 0   
1                      0                                 0   
2                      0                                 0   
3                      0                                 0   
4                      0                                 0   
..                   ...                               ...   
991                    0                                 0   
992                    0                                 0   
993                    0                                 0   
994                    0                                 0   
995                    0                                 0   

     Count of Clickbait Phrases  
0                             0  
1                             0  
2                             0  
3                             0  
4                             0  
..                          ...  
991                           0  
992                           0  
993                           0  
994                           0  
995                           0  

[996 rows x 6 columns]

In [6]:
newdf.to_csv('count_of_dicts_for_bollybait.csv')

In [7]:
#from google_trans_new import google_translator  
from emoji import UNICODE_EMOJI
import string
#translator = google_translator() 

if_numbers_present = list()
if_questions_present = list()
counts_of_emojis = list()
relative_freqs_of_caps = list()
counts_of_punct = list()

for index, row in data.iterrows():
    title = row['Translated Title']
    #title = translator.translate(titlex,lang_tgt='en')
    
    if_number_present = 0
    if_question_present = 0
    count_of_emojis = 0
    relative_freq_of_caps = 0
    count_of_punct = 0
    
    if any(char.isdigit() for char in title):
        if_number_present = 1
    else:
        if_number_present = 0
    if_numbers_present.append(if_number_present)
        
    if '?' in title:
        if_question_present = 1
    else:
        if_question_present = 0
    if_questions_present.append(if_question_present)
        
    for emoji in UNICODE_EMOJI:
        if emoji in title:
            count_of_emojis = count_of_emojis + 1
    counts_of_emojis.append(count_of_emojis)
        
    count_caps = 0
    count_lows = 0
    for i in title:
        if(i.islower()):
            count_lows = count_lows + 1
        elif(i.isupper()):
            count_caps = count_caps + 1
    if count_caps + count_lows == 0:
        relative_freq_of_caps = 0
    else:
        relative_freq_of_caps = count_caps/(count_caps + count_lows)
    relative_freqs_of_caps.append(relative_freq_of_caps)
        
    count = lambda l1,l2: sum([1 for x in l1 if x in l2])    
    count_of_punct = count(title,set(string.punctuation)) 
    counts_of_punct.append(count_of_punct)

dict1 = {'If Number Present' : if_numbers_present, 'If a Question' : if_questions_present, 'Count of Emojis' : counts_of_emojis, 'Relative Freq of Caps' : relative_freqs_of_caps, 'Count of Punctuation Marks' : counts_of_punct}
newdf1 = pd.DataFrame(dict1)
newdf1.head()

If Number Present  If a Question  Count of Emojis  Relative Freq of Caps  \
0                  0              0                1               0.156250   
1                  1              0                0               0.178571   
2                  0              0                0               0.157895   
3                  0              0                1               0.141026   
4                  0              0                2               0.066667   

   Count of Punctuation Marks  
0                           4  
1                           4  
2                           3  
3                           6  
4                           1

In [8]:
newdf1.to_csv('lexical_for_bollybait.csv')

In [9]:
from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer 
sid_obj = SentimentIntensityAnalyzer() 

positive_sentiment = list()
negative_sentiment = list()
neutral_sentiment = list()
compounded_sentiment = list()

for index, row in data.iterrows():
    title = row['Translated Title'].lower()
    #title = translator.translate(titlex,lang_tgt='en') 
    sentiment_dict = sid_obj.polarity_scores(title) 
    negative_sentiment.append(sentiment_dict['neg'])
    positive_sentiment.append(sentiment_dict['pos'])
    neutral_sentiment.append(sentiment_dict['neu'])
    compounded_sentiment.append(sentiment_dict['compound'])
    
dict2 = {'Positive Sentiment' : positive_sentiment, 'Negative Sentiment' : negative_sentiment, 'Neutral Sentiment' : neutral_sentiment, 'Compounded Sentiment' : compounded_sentiment}
newdf2 = pd.DataFrame(dict2)
newdf2.head()

Positive Sentiment  Negative Sentiment  Neutral Sentiment  \
0                 0.0                 0.0                1.0   
1                 0.0                 0.0                1.0   
2                 0.0                 0.0                1.0   
3                 0.0                 0.0                1.0   
4                 0.0                 0.0                1.0   

   Compounded Sentiment  
0                   0.0  
1                   0.0  
2                   0.0  
3                   0.0  
4                   0.0

In [10]:
newdf2.to_csv('sentiment_for_bollybait.csv')

In [11]:
import pandas as pd
import numpy as np
import torch
from transformers import BertTokenizer, BertModel
import matplotlib.pyplot as plt
tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
model = BertModel.from_pretrained('bert-base-uncased',output_hidden_states = True)

In [17]:
#from google_trans_new import google_translator 
import string
#translator = google_translator() 
mat = np.empty((0,768), int)
i = 0
for index, row in data.iterrows():
    i = i+1
    print(i)
    title = row['Translated Title']
    #title = translator.translate(titlex,lang_tgt='en')
    title = title.translate(str.maketrans('', '', string.punctuation))
    text = title.strip("[]")
    marked_text = "[CLS] " + text + " [SEP]"
    tokenized_text = tokenizer.tokenize(marked_text)
    indexed_tokens = tokenizer.convert_tokens_to_ids(tokenized_text)
    segments_ids = [1] * len(tokenized_text)
    tokens_tensor = torch.tensor([indexed_tokens])
    segments_tensors = torch.tensor([segments_ids])
    with torch.no_grad():
        outputs = model(tokens_tensor, segments_tensors)
        hidden_states = outputs[2]
    token_vecs = hidden_states[-2][0]
    sentence_embedding = torch.mean(token_vecs, dim = 0)
    rowx = np.reshape(sentence_embedding,(1, 768))
    mat = np.append(mat, rowx, axis = 0)

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199
200
201
202
203
204
205
206
207
208
209
210
211
212
213
214
215
216
217
218
219
220
221
222
223
224
225
226
227
228
229
230
231
232
233
234
235
236
237
238
239
240
241
242
243
244
245
246
247
248
249
250
251
252
253
254
255
256
257
258
259
260
261
262
263
264
265
266
267
268
269
270
271
272
273
274
275
276
277


In [18]:
import pandas as pd
df_bert = pd.DataFrame(mat)
df_bert.to_csv('bert_for_bollybait.csv')

from keras.preprocessing import image
from keras.applications.vgg16 import VGG16
from keras.applications.vgg16 import preprocess_input
from keras.applications.resnet50 import ResNet50
from keras.models import Model
import keras
import numpy as np
from keras.models import Sequential
from keras.layers import Dense, Dropout, Flatten
model = Sequential()
model.add(ResNet50(include_top=False, weights='imagenet', input_shape=(224,224,3), pooling = 'avg'))
model.summary()

i = 0
mat = np.empty((0,2048), int)
for index, row in data.iterrows():
    i = i + 1
    print(i)
    name_tag = row['video_id']
    folder_path = "/Users/peyamowar/Downloads/clickbait_youtube/" + name_tag + "/"
    try:
        img_path = "https://img.youtube.com/vi/" + name_tag + "/default.jpg"
        img = image.load_img(img_path, target_size=(224, 224))
        img_data = image.img_to_array(img)
        img_data = np.expand_dims(img_data, axis=0)
        img_data = preprocess_input(img_data)
        vgg16_feature = model.predict(img_data)
        mat = np.append(mat, vgg16_feature, axis = 0)

In [12]:
import pandas as pd
data1 = pd.read_csv("mvd/bert_for_mvd.csv")
data1 = data1.drop(columns = ['Unnamed: 0'])
data2 = pd.read_csv("mvd/count_of_dicts_for_mvd.csv")
data2 = data2.drop(columns = ['Unnamed: 0'])
data3 = pd.read_csv("mvd/lexical_for_mvd.csv")
data3 = data3.drop(columns = ['Unnamed: 0'])
data4 = pd.read_csv("mvd/sentiment_for_mvd.csv")
data4 = data4.drop(columns = ['Unnamed: 0'])
data5 = pd.read_csv("mvd/MVD_dataset_raw.csv")
#data5 = data5.drop(columns = ['Unnamed: 0'])

In [13]:
data1

0         1         2         3         4         5         6  \
0    0.656415 -0.588487  0.255769 -0.457658  0.070014 -0.430282  0.208616   
1    0.366475 -0.105111 -0.511447 -0.280992  0.036200 -0.419664  0.577570   
2    0.442341 -0.153139  0.180063 -0.506507  0.685487 -0.694880  0.302318   
3    0.189013 -0.487068 -0.017525 -0.428642  0.460981 -0.412667  0.287481   
4    0.309474 -0.757419 -0.338226 -0.159227  0.707320 -0.499518  0.264550   
..        ...       ...       ...       ...       ...       ...       ...   
987 -0.188280 -0.076149  0.133796 -0.753641  0.576994 -0.105465  0.646150   
988  0.142554 -0.599347  0.047055 -0.120339  0.484493 -0.286151 -0.189977   
989  0.230374 -0.681157  0.077860 -0.531160  0.515146 -0.507808  0.250623   
990 -0.005640 -0.443048 -0.027411 -0.430937  0.478407 -0.239432  0.100175   
991  0.166836 -0.238430 -0.166621 -0.624788  0.364746  0.002239  0.324703   

            7         8         9  ...       758       759       760  \
0    0.499660 -0.069495  0.193156  ... -0.017260  0.382134  0.727069   
1    0.173368 -0.598301  0.010184  ... -0.412046  0.464942  0.401888   
2    0.311490 -0.100956  0.340628  ... -0.576259 -0.060874  0.052163   
3    0.204468 -0.091526  0.142367  ... -0.006079 -0.215280  0.561446   
4    0.264980  0.061713  0.041802  ... -0.052071 -0.011489  0.406658   
..        ...       ...       ...  ...       ...       ...       ...   
987  0.314543  0.011651  0.218404  ... -0.627454  0.253633  0.341572   
988  0.274936 -0.436062  0.490000  ... -0.377252  0.181398  0.294635   
989  0.199079 -0.494489  0.335102  ... -0.333171  0.287498  0.342340   
990  0.254166 -0.252279  0.371059  ... -0.517625 -0.016664  0.278496   
991  0.700817 -0.175273  0.167172  ... -0.382226  0.344324  0.113995   

          761       762       763       764       765       766       767  
0    0.295227  0.288204  0.462131 -0.736290  0.025006 -0.082148 -0.326409  
1   -0.054143  0.609790  0.307075 -0.295643  0.056607 -0.231181 -0.512396  
2    0.054124  0.027412  0.747301 -0.574692 -0.150274  0.450653 -0.352937  
3   -0.428723  0.152730  0.650967  0.309384 -0.665767 -0.086792 -0.305832  
4   -0.461706 -0.244169  0.538682  0.220659 -0.174435  0.297273  0.273130  
..        ...       ...       ...       ...       ...       ...       ...  
987 -0.185746 -0.351311  0.697531 -0.402355 -0.300941  0.252272 -0.104030  
988  0.105738  0.088526  0.436400 -0.376651 -0.052997 -0.051290  0.215017  
989  0.002681  0.214874  0.384807 -0.142429 -0.063234 -0.111564 -0.395111  
990  0.043123 -0.079042  0.601956 -0.548550 -0.360947  0.349151  0.238950  
991 -0.263528  0.262136  0.551127 -0.332009 -0.144499  0.374626  0.142615  

[992 rows x 768 columns]

In [14]:
data2

Count of Celebrity Mentions  Count of Slangs  Count of Porn Words  \
0                              1                0                    0   
1                              1                0                    0   
2                              1                0                    0   
3                              0                0                    0   
4                              0                0                    1   
..                           ...              ...                  ...   
987                            1                0                    0   
988                            1                0                    0   
989                            1                0                    0   
990                            1                0                    0   
991                            1                0                    0   

     Count of Celebrity Related Words  Count of Clickbait Phrases  
0                                   3                           1  
1                                   1                           0  
2                                   0                           0  
3                                   0                           0  
4                                   0                           0  
..                                ...                         ...  
987                                 0                           0  
988                                 0                           0  
989                                 0                           0  
990                                 0                           0  
991                                 0                           0  

[992 rows x 5 columns]

In [15]:
data3

If Number Present  If a Question  Count of Emojis  Relative Freq of Caps  \
0                    1              0                1               0.271186   
1                    0              0                1               0.133333   
2                    0              0                1               0.102941   
3                    1              0                0               0.347826   
4                    0              0                0               0.129032   
..                 ...            ...              ...                    ...   
987                  0              0                1               0.125000   
988                  0              0                1               0.212766   
989                  0              0                1               0.171875   
990                  0              0                1               0.214286   
991                  0              0                1               0.204545   

     Count of Punctuation Marks  
0                             4  
1                             0  
2                             1  
3                             0  
4                             0  
..                          ...  
987                           0  
988                           1  
989                           3  
990                           3  
991                           2  

[992 rows x 5 columns]

In [16]:
data4

Positive Sentiment  Negative Sentiment  Neutral Sentiment  \
0                 0.000               0.389              0.611   
1                 0.000               0.796              0.204   
2                 0.000               0.000              1.000   
3                 0.000               0.000              1.000   
4                 0.000               0.367              0.633   
..                  ...                 ...                ...   
987               0.187               0.000              0.813   
988               0.000               0.281              0.719   
989               0.184               0.000              0.816   
990               0.000               0.000              1.000   
991               0.000               0.244              0.756   

     Compounded Sentiment  
0                 -0.7906  
1                 -0.5994  
2                  0.0000  
3                  0.0000  
4                 -0.4404  
..                    ...  
987                0.3182  
988               -0.5994  
989                0.4019  
990                0.0000  
991               -0.4404  

[992 rows x 4 columns]

In [17]:
data5

Title     video_id  \
0    Sad News For Dharmendra Fans, Dharmendra Death...  to8mQVw1Lso   
1                                     Dharmendra Death  rxTuq8gymc0   
2    Abhishek Bachchan and Shweta Bachchan in Koffe...  4N3hPrJLbEw   
3                     How To Get 6 Pack Abs In One Day  CfvUr6WAEWM   
4                 Dirty Secrets of the Bachchan Family  i1rlytoxQ1o   
..                                                 ...          ...   
987  Aishwarya accepts ‘Khajur’ as her son – The Ka...  mu1LmpE_rjE   
988  Cold War Between Lottery And Sarla - The Kapil...  L2rPcFB-bKg   
989  England Queen's Wedding Shopping | Sargun Meht...  EPq2r8hUCBM   
990  Kapil Copies Dr. Gulati's Look - The Kapil Sha...  gPuxJYAuT_g   
991  Rajesh Arora Hates Rinku's Face - The Kapil Sh...  DHJaC6tArpY   

         LABEL      Topic  
0    Clickbait      Death  
1    Clickbait      Death  
2    Clickbait  celebrity  
3    Clickbait     Health  
4    Clickbait  celebrity  
..         ...        ...  
987       Real        NaN  
988       Real        NaN  
989       Real        NaN  
990       Real        NaN  
991       Real        NaN  

[992 rows x 4 columns]

In [19]:
frames = [data5,data1,data2,data3,data4]
result = pd.concat(frames, axis = 1)
result

Title     video_id  \
0    Sad News For Dharmendra Fans, Dharmendra Death...  to8mQVw1Lso   
1                                     Dharmendra Death  rxTuq8gymc0   
2    Abhishek Bachchan and Shweta Bachchan in Koffe...  4N3hPrJLbEw   
3                     How To Get 6 Pack Abs In One Day  CfvUr6WAEWM   
4                 Dirty Secrets of the Bachchan Family  i1rlytoxQ1o   
..                                                 ...          ...   
987  Aishwarya accepts ‘Khajur’ as her son – The Ka...  mu1LmpE_rjE   
988  Cold War Between Lottery And Sarla - The Kapil...  L2rPcFB-bKg   
989  England Queen's Wedding Shopping | Sargun Meht...  EPq2r8hUCBM   
990  Kapil Copies Dr. Gulati's Look - The Kapil Sha...  gPuxJYAuT_g   
991  Rajesh Arora Hates Rinku's Face - The Kapil Sh...  DHJaC6tArpY   

         LABEL      Topic         0         1         2         3         4  \
0    Clickbait      Death  0.656415 -0.588487  0.255769 -0.457658  0.070014   
1    Clickbait      Death  0.366475 -0.105111 -0.511447 -0.280992  0.036200   
2    Clickbait  celebrity  0.442341 -0.153139  0.180063 -0.506507  0.685487   
3    Clickbait     Health  0.189013 -0.487068 -0.017525 -0.428642  0.460981   
4    Clickbait  celebrity  0.309474 -0.757419 -0.338226 -0.159227  0.707320   
..         ...        ...       ...       ...       ...       ...       ...   
987       Real        NaN -0.188280 -0.076149  0.133796 -0.753641  0.576994   
988       Real        NaN  0.142554 -0.599347  0.047055 -0.120339  0.484493   
989       Real        NaN  0.230374 -0.681157  0.077860 -0.531160  0.515146   
990       Real        NaN -0.005640 -0.443048 -0.027411 -0.430937  0.478407   
991       Real        NaN  0.166836 -0.238430 -0.166621 -0.624788  0.364746   

            5  ...  Count of Clickbait Phrases  If Number Present  \
0   -0.430282  ...                           1                  1   
1   -0.419664  ...                           0                  0   
2   -0.694880  ...                           0                  0   
3   -0.412667  ...                           0                  1   
4   -0.499518  ...                           0                  0   
..        ...  ...                         ...                ...   
987 -0.105465  ...                           0                  0   
988 -0.286151  ...                           0                  0   
989 -0.507808  ...                           0                  0   
990 -0.239432  ...                           0                  0   
991  0.002239  ...                           0                  0   

     If a Question  Count of Emojis  Relative Freq of Caps  \
0                0                1               0.271186   
1                0                1               0.133333   
2                0                1               0.102941   
3                0                0               0.347826   
4                0                0               0.129032   
..             ...              ...                    ...   
987              0                1               0.125000   
988              0                1               0.212766   
989              0                1               0.171875   
990              0                1               0.214286   
991              0                1               0.204545   

     Count of Punctuation Marks  Positive Sentiment  Negative Sentiment  \
0                             4               0.000               0.389   
1                             0               0.000               0.796   
2                             1               0.000               0.000   
3                             0               0.000               0.000   
4                             0               0.000               0.367   
..                          ...                 ...                 ...   
987                           0               0.187               0.000   
988                           1               0.000               0.281  

In [20]:
result.to_csv("mvd/MVD.csv")

In [12]:
import pandas as pd
data1 = pd.read_csv("bollybait/BollyBAIT.csv")
#data1 = data1.drop(columns = ['Unnamed: 0'])
data2 = pd.read_csv("bollybait/audio_vs_title_bollybait.csv")
data2 = data2.drop(columns = ['Unnamed: 0'])
data3 = pd.read_csv("bollybait/graph_cnn_bollybait.csv")
data3 = data3.drop(columns = ['Unnamed: 0'])
data4 = pd.read_csv("bollybait/resnet_for_thumbnail_bollybait.csv")
data4 = data4.drop(columns = ['Unnamed: 0'])
data5 = pd.read_csv("bollybait/thumbnail_vs_title_bollybait.csv")
data5 = data5.drop(columns = ['Unnamed: 0'])

In [13]:
data1 = data1.drop(columns = ['Unnamed: 0'])
data1

Title  \
0    Brahmastra Trailer | Ranbir Kapoor | Alia Bhat...   
1    83 Movie Trailer | Ranveer Singh | Deepika Pad...   
2    Bell Bottom Trailer | Akshay Kumar | Jackky  B...   
3    Mahabharat - Official Trailer | Aamir Khan | H...   
4    Nick Jonas की इस हरकत पर भड़की Priyanka Chopra,...   
..                                                 ...   
655  Good Newwz | Shoot Begins | Akshay Kumar | Kar...   
656  Nanhi Si Jaan | 2.0 (Hindi) | Rajinikanth | Ak...   
657  2.0 - Official Teaser [Hindi] | Rajinikanth | ...   
658             Madhuri's Bucket List - Part 1 | Hindi   
659  Watan ke aage kuch nahi | Raazi | Alia Bhatt |...   

                                      Translated Title     Video ID  \
0    Brahmastra Trailer | Ranbir Kapoor | Alia Bhat...  zR8x5FAvlPU   
1    83 Movie Trailer | Ranveer Singh | Deepika Pad...  STL-ueSScRI   
2    Bell Bottom Trailer | Akshay Kumar | Jackky Bh...  bJ-1QOOett0   
3    Mahabharat - Official Trailer | Aamir Khan | H...  qfr0NUD3FTU   
4    Nick erupted on the movement of Jonas Priyanka...  D_TP0L-s4V0   
..                                                 ...          ...   
655  Good Newwz | Shoot Begins | Akshay Kumar | Kar...  tOWpOL4ZgSw   
656  Nanhi Si Jaan | 2.0 (Hindi). Rajinikanth | Aks...  MBrAY6GgU5k   
657  2.0 - Official Teaser [Hindi] | Rajinikanth | ...  owPuQjInzO8   
658             Madhuri's Bucket List - Part 1 | Hindi  Es2bli0uSYY   
659  Watan Ke Aage Kuch Nahi | Raazi | Alia Bhatt |...  FgHKdoYpGpI   

                                       Video URL       Channel Title  \
0    https://www.youtube.com/watch?v=zR8x5FAvlPU        PR Bollywood   
1    https://www.youtube.com/watch?v=STL-ueSScRI        PR Bollywood   
2    https://www.youtube.com/watch?v=bJ-1QOOett0        PR Bollywood   
3    https://www.youtube.com/watch?v=qfr0NUD3FTU        PR Bollywood   
4    https://www.youtube.com/watch?v=D_TP0L-s4V0        PR Bollywood   
..                                           ...                 ...   
655  https://www.youtube.com/watch?v=tOWpOL4ZgSw  Dharma Productions   
656  https://www.youtube.com/watch?v=MBrAY6GgU5k  Dharma Productions   
657  https://www.youtube.com/watch?v=owPuQjInzO8  Dharma Productions   
658  https://www.youtube.com/watch?v=Es2bli0uSYY  Dharma Productions   
659  https://www.youtube.com/watch?v=FgHKdoYpGpI  Dharma Productions   

    Misleading Video False Promises  Exaggerated Video Spam Content   \
0                 No             Yes               Yes            No   
1                 No             Yes               Yes            No   
2                Yes             Yes               Yes            No   
3                 No             Yes               Yes            No   
4                Yes             Yes               Yes           Yes   
..               ...             ...               ...           ...   
655               No              No                No            No   
656               No              No                No            No   
657               No              No                No            No   
658               No              No                No            No   
659               No              No                No            No   

    Exploits Curiosity Gap   ... 2038resnet  2039resnet  2040resnet  \
0                        No  ...   0.284980    1.428380    1.179158   
1                        No  ...   0.314725    0.190759    1.376088   
2                        No  ...   1.010923    0.072747    0.711391   
3                        No  ...   0.696061    0.408475    1.596582   
4                        No  ...   0.602495    0.062648    0.410917   
..                      ...  ...        ...         ...         ...   
655                      No  ...   1.743024    0.465506    0.514298   
656                      No  ...   0.015850    2.930738    1.730189   
657                      No  ...   2.581937    0.595867    1.347908   
658                      No  ...   0.021248    0.073434   

In [61]:
data2

Video ID  Title vs Audio 1  Title vs Audio 2  Title vs Audio 3
0    zR8x5FAvlPU              0.00              1.97            100.00
1    STL-ueSScRI             11.18             60.82             86.52
2    bJ-1QOOett0             16.12             44.79             89.34
3    qfr0NUD3FTU              8.13             29.83            100.00
4    D_TP0L-s4V0             36.51             56.19            100.00
..           ...               ...               ...               ...
883  tOWpOL4ZgSw              0.00             52.27            100.00
884  MBrAY6GgU5k              0.00             44.94            100.00
885  owPuQjInzO8              0.00             12.69            100.00
886  Es2bli0uSYY              7.40             47.90            100.00
887  FgHKdoYpGpI              6.90             54.64             91.09

[888 rows x 4 columns]

In [14]:
data3
data3 = data3.drop(columns = ['Index'])
data3 = data3.add_suffix('graphcnn')
data3 = data3.rename(columns = {'Movie IDgraphcnn' : 'Movie ID'})
data3

0graphcnn  1graphcnn  2graphcnn  3graphcnn  4graphcnn  5graphcnn  \
0          0.0        0.0        0.0        0.0        0.0   0.000216   
1          0.0        0.0        0.0        0.0        0.0   0.000009   
2          0.0        0.0        0.0        0.0        0.0   0.000000   
3          0.0        0.0        0.0        0.0        0.0   0.000000   
4          0.0        0.0        0.0        0.0        0.0   0.000974   
..         ...        ...        ...        ...        ...        ...   
740        0.0        0.0        0.0        0.0        0.0   0.002374   
741        0.0        0.0        0.0        0.0        0.0   0.000000   
742        0.0        0.0        0.0        0.0        0.0   0.000000   
743        0.0        0.0        0.0        0.0        0.0   0.000000   
744        0.0        0.0        0.0        0.0        0.0   0.000367   

     6graphcnn  7graphcnn  8graphcnn  9graphcnn  10graphcnn  11graphcnn  \
0          0.0   0.000000        0.0   0.000000         0.0         0.0   
1          0.0   0.000000        0.0   0.000000         0.0         0.0   
2          0.0   0.000000        0.0   0.000000         0.0         0.0   
3          0.0   0.000922        0.0   0.001092         0.0         0.0   
4          0.0   0.000000        0.0   0.000000         0.0         0.0   
..         ...        ...        ...        ...         ...         ...   
740        0.0   0.000000        0.0   0.000000         0.0         0.0   
741        0.0   0.005233        0.0   0.000000         0.0         0.0   
742        0.0   0.000000        0.0   0.000000         0.0         0.0   
743        0.0   0.005077        0.0   0.000000         0.0         0.0   
744        0.0   0.000000        0.0   0.000000         0.0         0.0   

     12graphcnn  13graphcnn  14graphcnn  15graphcnn     Movie ID  
0           0.0         0.0         0.0         0.0  zR8x5FAvlPU  
1           0.0         0.0         0.0         0.0  STL-ueSScRI  
2           0.0         0.0         0.0         0.0  bJ-1QOOett0  
3           0.0         0.0         0.0         0.0  qfr0NUD3FTU  
4           0.0         0.0         0.0         0.0  D_TP0L-s4V0  
..          ...         ...         ...         ...          ...  
740         0.0         0.0         0.0         0.0  MBrAY6GgU5k  
741         0.0         0.0         0.0         0.0  owPuQjInzO8  
742         0.0         0.0         0.0         0.0  Es2bli0uSYY  
743         0.0         0.0         0.0         0.0  FgHKdoYpGpI  
744         0.0         0.0         0.0         0.0  PJ-mIEZMJes  

[745 rows x 17 columns]

In [63]:
#data4 = pd.read_csv("mvd/resnet_for_thumbnail_mvd.csv")
#data4 = data4.drop(columns = ['Unnamed: 0'])
data4 = data4.add_suffix('resnet')
data4 = data4.rename(columns = {'Movie IDresnet' : 'Movie ID'})
data4

0resnet   1resnet   2resnet   3resnet   4resnet   5resnet   6resnet  \
0    0.186488  0.654522  0.015405  1.468111  3.018486  0.345861  0.554258   
1    0.174984  1.158020  0.119610  1.257308  1.903232  0.974811  0.251946   
2    0.433122  0.162114  0.266389  0.792907  2.171352  0.534331  0.198576   
3    0.215302  1.122640  0.010135  0.408526  1.493597  0.788077  0.500551   
4    0.221501  2.145802  0.000000  3.219835  1.394504  1.416848  0.865140   
..        ...       ...       ...       ...       ...       ...       ...   
740  0.040095  0.355303  0.043459  0.950834  2.597373  0.510572  0.711961   
741  1.018428  2.808853  0.220803  0.782658  2.502432  1.254771  0.181124   
742  0.040714  0.697384  0.000000  2.006764  0.625356  0.000000  0.173528   
743  0.115539  1.467525  0.000000  0.992333  1.656572  0.045711  0.060132   
744  0.000000  0.366458  0.021986  1.054767  0.841519  0.000000  0.135761   

      7resnet   8resnet   9resnet  ...  2039resnet  2040resnet  2041resnet  \
0    0.006540  2.777847  0.000000  ...    1.428380    1.179158    3.490497   
1    0.159007  6.775068  0.003984  ...    0.190759    1.376088    2.323992   
2    0.848471  4.107092  0.018591  ...    0.072747    0.711391    0.015284   
3    0.846385  2.214705  0.003534  ...    0.408475    1.596582    1.728786   
4    0.677972  2.043006  0.161614  ...    0.062648    0.410917    1.891637   
..        ...       ...       ...  ...         ...         ...         ...   
740  0.406390  1.826892  0.013691  ...    2.930738    1.730189    1.612979   
741  0.678292  2.233051  0.003673  ...    0.595867    1.347908    1.152688   
742  0.402590  0.424832  0.581553  ...    0.073434    0.665139    5.099169   
743  0.221198  2.358593  0.000000  ...    0.290015    1.071062    3.769875   
744  0.138933  0.516578  0.000000  ...    0.080660    0.561831    0.093123   

     2042resnet  2043resnet  2044resnet  2045resnet  2046resnet  2047resnet  \
0      0.604142    0.678579    0.613538    0.134320    0.273965    0.417604   
1      0.614419    0.186999    0.720658    0.168697    0.682503    0.377021   
2      0.000000    0.339965    0.001930    0.035980    0.241261    0.412969   
3      1.728387    0.622000    0.782353    0.015879    1.023183    0.034589   
4      2.436680    0.065670    1.112929    0.059084    0.803206    0.378065   
..          ...         ...         ...         ...         ...         ...   
740    0.713008    0.160858    0.233176    0.000000    0.342393    0.000000   
741    0.161927    0.338069    0.705395    0.000000    0.660729    0.001487   
742    0.251920    0.000000    0.530523    0.286724    0.061793    1.265754   
743    0.292817    0.062635    0.629411    0.003198    0.084911    0.308729   
744    0.840210    0.000000    0.348033    0.203431    0.194597    0.009497   

        Movie ID  
0    zR8x5FAvlPU  
1    STL-ueSScRI  
2    bJ-1QOOett0  
3    qfr0NUD3FTU  
4    D_TP0L-s4V0  
..           ...  
740  MBrAY6GgU5k  
741  owPuQjInzO8  
742  Es2bli0uSYY  
743  FgHKdoYpGpI  
744  PJ-mIEZMJes  

[745 rows x 2049 columns]

In [67]:
data5
data5 = data5.rename(columns = {'Video ID' : 'Video ID1'})
data5

Video ID1  Title vs Thumbnail 1  Title vs Thumbnail 2  \
0    zR8x5FAvlPU                   0.0                 -0.22   
1    STL-ueSScRI                   0.0                 13.53   
2    bJ-1QOOett0                   0.0                 30.40   
3    qfr0NUD3FTU                   0.0                -10.95   
4    D_TP0L-s4V0                   0.0                 17.71   
..           ...                   ...                   ...   
991  MBrAY6GgU5k                   0.0                  8.15   
992  owPuQjInzO8                   0.0                 13.49   
993  Es2bli0uSYY                   0.0                 44.48   
994  FgHKdoYpGpI                   0.0                 -3.41   
995  PJ-mIEZMJes                   0.0                  4.65   

     Title vs Thumbnail 3  
0                   100.0  
1                   100.0  
2                   100.0  
3                   100.0  
4                   100.0  
..                    ...  
991                 100.0  
992                 100.0  
993                 100.0  
994                 100.0  
995                 100.0  

[996 rows x 4 columns]

In [8]:
frames = [data1,data3]
result = pd.concat(frames, axis = 1)
result = result.drop(columns = ['Indexgraphcnn', 'Movie ID'])
result

Title  \
0    Brahmastra Trailer | Ranbir Kapoor | Alia Bhat...   
1    83 Movie Trailer | Ranveer Singh | Deepika Pad...   
2    Bell Bottom Trailer | Akshay Kumar | Jackky  B...   
3    Mahabharat - Official Trailer | Aamir Khan | H...   
4    Nick Jonas की इस हरकत पर भड़की Priyanka Chopra,...   
..                                                 ...   
740                                                NaN   
741                                                NaN   
742                                                NaN   
743                                                NaN   
744                                                NaN   

                                      Translated Title     Video ID  \
0    Brahmastra Trailer | Ranbir Kapoor | Alia Bhat...  zR8x5FAvlPU   
1    83 Movie Trailer | Ranveer Singh | Deepika Pad...  STL-ueSScRI   
2    Bell Bottom Trailer | Akshay Kumar | Jackky Bh...  bJ-1QOOett0   
3    Mahabharat - Official Trailer | Aamir Khan | H...  qfr0NUD3FTU   
4    Nick erupted on the movement of Jonas Priyanka...  D_TP0L-s4V0   
..                                                 ...          ...   
740                                                NaN          NaN   
741                                                NaN          NaN   
742                                                NaN          NaN   
743                                                NaN          NaN   
744                                                NaN          NaN   

                                       Video URL Channel Title  \
0    https://www.youtube.com/watch?v=zR8x5FAvlPU  PR Bollywood   
1    https://www.youtube.com/watch?v=STL-ueSScRI  PR Bollywood   
2    https://www.youtube.com/watch?v=bJ-1QOOett0  PR Bollywood   
3    https://www.youtube.com/watch?v=qfr0NUD3FTU  PR Bollywood   
4    https://www.youtube.com/watch?v=D_TP0L-s4V0  PR Bollywood   
..                                           ...           ...   
740                                          NaN           NaN   
741                                          NaN           NaN   
742                                          NaN           NaN   
743                                          NaN           NaN   
744                                          NaN           NaN   

    Misleading Video False Promises  Exaggerated Video Spam Content   \
0                 No             Yes               Yes            No   
1                 No             Yes               Yes            No   
2                Yes             Yes               Yes            No   
3                 No             Yes               Yes            No   
4                Yes             Yes               Yes           Yes   
..               ...             ...               ...           ...   
740              NaN             NaN               NaN           NaN   
741              NaN             NaN               NaN           NaN   
742              NaN             NaN               NaN           NaN   
743              NaN             NaN               NaN           NaN   
744              NaN             NaN               NaN           NaN   

    Exploits Curiosity Gap   ... 6graphcnn  7graphcnn  8graphcnn  9graphcnn  \
0                        No  ...       0.0   0.000000        0.0   0.000000   
1                        No  ...       0.0   0.000000        0.0   0.000000   
2                        No  ...       0.0   0.000000        0.0   0.000000   
3                        No  ...       0.0   0.000922        0.0   0.001092   
4                        No  ...       0.0   0.000000        0.0   0.000000   
..                      ...  ...       ...        ...        ...        ...   
740                     NaN  ...       0.0   0.000000        0.0   0.000000   
741                     NaN  ...       0.0   0.005233        0.0   0.000000   
742                     NaN  ...       0.0   0.000000        0.0   0.000000   
743                     NaN  ...       0.0   0.005

In [71]:
#data2 = data2.rename(columns = {'Video ID' : 'video_id'})
result1 = result.join(data2.set_index('Video ID'), how='inner', on='Video ID')
result1 = result1.drop_duplicates(subset='Video ID', keep="last")
result = result1
#prresult1int(type(result['video_id'][0]))

In [15]:
data3 = data3.rename(columns = {'Movie ID' : 'Video ID'})
result1 = data1.join(data3.set_index('Video ID'), how='inner', on='Video ID')
result1 = result1.drop_duplicates(subset='Video ID', keep="last")
#result1
result = result1
#print(type(result['video_id'][0]))

In [16]:
result

Title  \
0    Brahmastra Trailer | Ranbir Kapoor | Alia Bhat...   
1    83 Movie Trailer | Ranveer Singh | Deepika Pad...   
2    Bell Bottom Trailer | Akshay Kumar | Jackky  B...   
3    Mahabharat - Official Trailer | Aamir Khan | H...   
4    Nick Jonas की इस हरकत पर भड़की Priyanka Chopra,...   
..                                                 ...   
655  Good Newwz | Shoot Begins | Akshay Kumar | Kar...   
656  Nanhi Si Jaan | 2.0 (Hindi) | Rajinikanth | Ak...   
657  2.0 - Official Teaser [Hindi] | Rajinikanth | ...   
658             Madhuri's Bucket List - Part 1 | Hindi   
659  Watan ke aage kuch nahi | Raazi | Alia Bhatt |...   

                                      Translated Title     Video ID  \
0    Brahmastra Trailer | Ranbir Kapoor | Alia Bhat...  zR8x5FAvlPU   
1    83 Movie Trailer | Ranveer Singh | Deepika Pad...  STL-ueSScRI   
2    Bell Bottom Trailer | Akshay Kumar | Jackky Bh...  bJ-1QOOett0   
3    Mahabharat - Official Trailer | Aamir Khan | H...  qfr0NUD3FTU   
4    Nick erupted on the movement of Jonas Priyanka...  D_TP0L-s4V0   
..                                                 ...          ...   
655  Good Newwz | Shoot Begins | Akshay Kumar | Kar...  tOWpOL4ZgSw   
656  Nanhi Si Jaan | 2.0 (Hindi). Rajinikanth | Aks...  MBrAY6GgU5k   
657  2.0 - Official Teaser [Hindi] | Rajinikanth | ...  owPuQjInzO8   
658             Madhuri's Bucket List - Part 1 | Hindi  Es2bli0uSYY   
659  Watan Ke Aage Kuch Nahi | Raazi | Alia Bhatt |...  FgHKdoYpGpI   

                                       Video URL       Channel Title  \
0    https://www.youtube.com/watch?v=zR8x5FAvlPU        PR Bollywood   
1    https://www.youtube.com/watch?v=STL-ueSScRI        PR Bollywood   
2    https://www.youtube.com/watch?v=bJ-1QOOett0        PR Bollywood   
3    https://www.youtube.com/watch?v=qfr0NUD3FTU        PR Bollywood   
4    https://www.youtube.com/watch?v=D_TP0L-s4V0        PR Bollywood   
..                                           ...                 ...   
655  https://www.youtube.com/watch?v=tOWpOL4ZgSw  Dharma Productions   
656  https://www.youtube.com/watch?v=MBrAY6GgU5k  Dharma Productions   
657  https://www.youtube.com/watch?v=owPuQjInzO8  Dharma Productions   
658  https://www.youtube.com/watch?v=Es2bli0uSYY  Dharma Productions   
659  https://www.youtube.com/watch?v=FgHKdoYpGpI  Dharma Productions   

    Misleading Video False Promises  Exaggerated Video Spam Content   \
0                 No             Yes               Yes            No   
1                 No             Yes               Yes            No   
2                Yes             Yes               Yes            No   
3                 No             Yes               Yes            No   
4                Yes             Yes               Yes           Yes   
..               ...             ...               ...           ...   
655               No              No                No            No   
656               No              No                No            No   
657               No              No                No            No   
658               No              No                No            No   
659               No              No                No            No   

    Exploits Curiosity Gap   ... 6graphcnn  7graphcnn  8graphcnn  9graphcnn  \
0                        No  ...       0.0   0.000000        0.0   0.000000   
1                        No  ...       0.0   0.000000        0.0   0.000000   
2                        No  ...       0.0   0.000000        0.0   0.000000   
3                        No  ...       0.0   0.000922        0.0   0.001092   
4                        No  ...       0.0   0.000000        0.0   0.000000   
..                      ...  ...       ...        ...        ...        ...   
655                      No  ...       0.0   0.000000        0.0   0.000000   
656                      No  ...       0.0   0.000000        0.0   0.000000   
657                      No  ...       0.0   0.005233    

In [75]:
data4 = data4.rename(columns = {'Movie ID' : 'Video ID'})
result1 = result.join(data4.set_index('Video ID'), how='inner', on='Video ID')
result1 = result1.drop_duplicates(subset='Video ID', keep="last")
result1
#print(type(result['video_id'][0]))

ValueError: columns overlap but no suffix specified: Index(['0resnet', '1resnet', '2resnet', '3resnet', '4resnet', '5resnet',
       '6resnet', '7resnet', '8resnet', '9resnet',
       ...
       '2038resnet', '2039resnet', '2040resnet', '2041resnet', '2042resnet',
       '2043resnet', '2044resnet', '2045resnet', '2046resnet', '2047resnet'],
      dtype='object', length=2048)

In [17]:
result.to_csv("bollybait/BollyBAIT.csv")